bugsnag-android-core/src/main/java/com/bugsnag/android/ReportDeliveryDelegate.java

```diff
@@ -0,0 +1,109 @@
+package com.bugsnag.android;
+
+import androidx.annotation.NonNull;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.RejectedExecutionException;
+
+class ReportDeliveryDelegate extends BaseObservable {
+
+    final Logger logger;
+    final EventStore eventStore;
+    final ImmutableConfig immutableConfig;
+    final BreadcrumbState breadcrumbState;
+
+    ReportDeliveryDelegate(Logger logger, EventStore eventStore,
+                           ImmutableConfig immutableConfig, BreadcrumbState breadcrumbState) {
+        this.logger = logger;
+        this.eventStore = eventStore;
+        this.immutableConfig = immutableConfig;
+        this.breadcrumbState = breadcrumbState;
+    }
+
+    void deliverEvent(@NonNull Event event) {
+        // Build the report
+        Report report = new Report(immutableConfig.getApiKey(), event);
+        Session session = event.getSession();
```

Comment:
Shouldn't this be set from `SessionTracker`'s current session?

We should also have assertions in the tests for this session to be incremented.

Reply:
My understanding is that the previous `currentSession` variable is equivalent to `event.getSession()` - and that therefore we can just remove that variable as its redundant.

Reply:
We have test coverage for session counts being incremented in `generateUnhandledReport` and  `generateHandledReport` - if more is required it'd be good to understand what needs adding.